src/components/MutationTableTabContent.tsx

```diff
@@ -0,0 +1,146 @@
+  variants: { query: VariantSelector; name: string; description: string }[];
+  dateRangeSelector: DateRangeSelector;
+  mode: 'Single' | 'CompareToBaseline';
+  baselineVariant?: VariantSelector;
+};
+
+export default function MutationTableTabContent({
+  mutationType,
+  variants,
+  locationSelector,
+  dateRangeSelector,
+  mode,
+  baselineVariant,
+}: MutationTableTabContentProps) {
+  const mutations = useQuery(
+    async signal => {
+      if (!variants) {
+        return undefined;
+      }
+      const data: MutationObject[] = [];
+
+      await Promise.allSettled(
+        variants.map(variant => {
+          const selector: LapisSelector = addDefaultHostAndQc({
+            location: locationSelector,
+            variant: variant.query,
+            samplingStrategy: SamplingStrategy.AllSamples,
+            dateRange: dateRangeSelector,
+          });
+
+          return Promise.all([
+            fetchSamplesCount(selector, signal),
+            MutationProportionData.fromApi(selector, 'aa', signal),
+            MutationProportionData.fromApi(selector, 'nuc', signal),
+          ]).then(async ([variantCount, aaMutationDataset, nucMutationDataset]) => {
+            data.push({
+              ...variant,
+              variantCount: variantCount,
+              aaMutations: aaMutationDataset.payload,
+              nucMutations: nucMutationDataset.payload,
+            });
+          });
+        })
+      );
+
+      return data;
+    },
+    [variants, locationSelector, dateRangeSelector]
+  );
+
+  const baselineMutations = useQuery(
+    signal => {
+      if (mode !== 'CompareToBaseline') {
+        return Promise.resolve({
+          aaMutations: [],
+          nucMutations: [],
+        });
+      }
+
+      const selector: LapisSelector = addDefaultHostAndQc({
+        location: locationSelector,
+        variant: baselineVariant,
+        samplingStrategy: SamplingStrategy.AllSamples,
+        dateRange: dateRangeSelector,
+      });
+
+      return Promise.all([
+        MutationProportionData.fromApi(selector, 'aa', signal),
+        MutationProportionData.fromApi(selector, 'nuc', signal),
+      ]).then(async ([aaMutationDataset, nucMutationDataset]) => {
+        return {
+          aaMutations: aaMutationDataset.payload,
+          nucMutations: nucMutationDataset.payload,
+        };
+      });
+    },
+    [baselineVariant, mode, locationSelector, dateRangeSelector]
+  );
+
+  const mutationTableData = useMemo(() => {
+    if (mutations.data) {
+      return mutations.data.map((variant, i) => {
+        return {
+          query: variant.query,
+          id: i,
+          name: variant.name.length > 0 ? variant.name : formatVariantDisplayName(variant.query),
+          variantCount: variant.variantCount,
+          aaMutations: variant.aaMutations,
+          nucMutations: variant.nucMutations,
+        };
+      });
+    }
+  }, [mutations]);
+
+  const baselineTableData = useMemo(() => {
```

Comment:
Why do we actually need a `useMemo` here? Couldn't we simply write `mode === 'CompareToBaseline' && baselineMutations.data && mutations.data ? baselineMutations.data : undefined`?

Reply:
Why do we need that check at all? Isn't is essentially the same check as line 132 below?